pkg/network/additional_networks.go

```diff
@@ -58,3 +62,140 @@ func validateRaw(conf *operv1.AdditionalNetworkDefinition) []error {
+	staticIPAMConfig := staticIPAMConfig{}
+	staticIPAMConfig.Type = "static"
+	for _, address := range conf.Addresses {
+		staticIPAMConfig.Addresses = append(staticIPAMConfig.Addresses, staticIPAMAddress{AddressStr: address.Address, Gateway: net.ParseIP(address.Gateway)})
+	}
+	for _, route := range conf.Routes {
+		_, dest, err := net.ParseCIDR(route.Destination)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to parse macvlan route")
+		}
+		staticIPAMConfig.Routes = append(staticIPAMConfig.Routes, &cnitypes.Route{Dst: *dest, GW: net.ParseIP(route.Gateway)})
+	}
+
+	if conf.DNS != nil {
+		staticIPAMConfig.DNS.Nameservers = append(staticIPAMConfig.DNS.Nameservers, conf.DNS.Nameservers...)
+		staticIPAMConfig.DNS.Domain = conf.DNS.Domain
+		staticIPAMConfig.DNS.Search = append(staticIPAMConfig.DNS.Search, conf.DNS.Search...)
+	}
+
+	jsonByte, err := json.Marshal(staticIPAMConfig)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create static ipam config")
+	}
+
+	return string(jsonByte), nil
+}
+
+// getIPAMConfigJSON generates IPAM CNI json config
+func getIPAMConfigJSON(conf *operv1.IPAMConfig) (string, error) {
+
+	if conf == nil || conf.Type == operv1.IPAMTypeDHCP {
+		// DHCP does not have additional config
+		return `{ "type": "dhcp" }`, nil
+	} else if conf.Type == operv1.IPAMTypeStatic {
+		staticIPAMConfig, err := getStaticIPAMConfigJSON(conf.StaticIPAMConfig)
+		return staticIPAMConfig, err
+	}
+
+	return "", errors.Errorf("failed to render IPAM JSON")
+}
+
+// renderSimpleMacvlanConfig returns the SimpleMacvlanConfig manifests
+func renderSimpleMacvlanConfig(conf *operv1.AdditionalNetworkDefinition, manifestDir string) ([]*uns.Unstructured, error) {
+	var err error
+	objs := []*uns.Unstructured{}
+
+	// render SimpleMacvlanConfig manifests
+	data := render.MakeRenderData()
+	data.Data["AdditionalNetworkName"] = conf.Name
+
+	if conf.SimpleMacvlanConfig == nil {
+		// no additional config, just fill default IPAM
+		data.Data["IPAMConfig"], err = getIPAMConfigJSON(nil)
+	} else {
+		macvlanConfig := conf.SimpleMacvlanConfig
+		data.Data["Master"] = macvlanConfig.Master
+
+		data.Data["IPAMConfig"], err = getIPAMConfigJSON(macvlanConfig.IPAMConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to render ipam config")
+		}
+
+		if macvlanConfig.Mode != "" {
+			// macvlan CNI only accepts mode in lowercase
+			data.Data["Mode"] = strings.ToLower(string(macvlanConfig.Mode))
+		}
+
+		if macvlanConfig.MTU != 0 {
+			data.Data["MTU"] = macvlanConfig.MTU
+		}
+	}
+
+	objs, err = render.RenderDir(filepath.Join(manifestDir, "network/additional-networks/simplemacvlan"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render simplemacvlan additional network")
+	}
+	return objs, nil
+}
+
+// validateIPAMConfig checks its IPAMConfig.
+func validateIPAMConfig(conf *operv1.IPAMConfig) []error {
+	out := []error{}
+
+	switch conf.Type {
+	case operv1.IPAMTypeDHCP:
+	case operv1.IPAMTypeStatic:
```

Comment:
Surely there is some validation you can do here? Check to see if supplied routes are valid IP addresses?

Reply:
added.